webrender_traits/src/display_list.rs

```diff
@@ -223,27 +223,128 @@ impl DisplayListBuilder {
         }
     }
 
+    // Gradients can be defined with stops outside the range of [0, 1]
+    // when this happens the gradient needs to be normalized by adjusting
+    // the gradient stops and gradient line into an equivalent gradient
+    // with stops in the range [0, 1]. this is done by moving the beginning
+    // of the gradient line to where stop[0] and the end of the gradient line
+    // to stop[n-1]. this function adjusts the stops in place, and returns
+    // the amount to adjust the gradient line start and stop
+    fn normalize_stops(stops: &mut Vec<GradientStop>,
```

Comment:
Ah, wait `normalize_stops` can in the bad case clear() and add() to the list. That's why it's a `Vec`.